src/python/pants/core/goals/test.py

```diff
@@ -949,7 +977,10 @@ async def run_tests(
         if result.result_metadata is None:
             # We end up here, e.g., if compilation failed during self-implemented test discovery.
             continue
-
+        if test_subsystem.report_test_result_source:
+            test_result_sources[result.addresses[0].spec] = result.result_metadata.source(
+                run_id
+            ).value
```

Comment:
Ah, one more thought: related to the "public API" comment above. The `ProcessResultMetadata.Source` enum values are now public API, so maybe that could either be:
1. documented in `process.py`
2. we separate the public reported value here from the underlying enum values (i.e. do an explicit mapping step from enum -> str that hard-codes the appropriate value for each enum value)

I'm inclined to 1. Thoughts?

Reply:
Thanks for thinking about this! I think 2 maybe an overkill, so I'm going with (1) by adding a docstring to the enum.